stdlib/multiprocessing/util.pyi

```diff
@@ -0,0 +1,122 @@
+import threading
+from _typeshed import Self, StrOrBytesPath
+from collections.abc import Callable, Iterable, Sequence
+from itertools import count
+from logging import Logger
+from typing import Any
+from weakref import WeakValueDictionary
+
+from . import process
+
+__all__ = [
+    "sub_debug",
+    "debug",
+    "info",
+    "sub_warning",
+    "get_logger",
+    "log_to_stderr",
+    "get_temp_dir",
+    "register_after_fork",
+    "is_exiting",
+    "Finalize",
+    "ForkAwareThreadLock",
+    "ForkAwareLocal",
+    "close_all_fds_except",
+    "SUBDEBUG",
+    "SUBWARNING",
+]
+
+NOTSET: int = ...
+SUBDEBUG: int = ...
+DEBUG: int = ...
+INFO: int = ...
+SUBWARNING: int = ...
+
+LOGGER_NAME: str = ...
+DEFAULT_LOGGING_FORMAT: str = ...
+
+_logger: Logger | None = ...
+_log_to_stderr: bool = ...
+
+def sub_debug(msg: object, *args: object) -> None: ...
+def debug(msg: object, *args: object) -> None: ...
+def info(msg: object, *args: object) -> None: ...
+def sub_warning(msg: object, *args: object) -> None: ...
+def get_logger() -> Logger: ...
+def log_to_stderr(level: int | None = ...) -> Logger: ...
+def _platform_supports_abstract_sockets() -> bool: ...
+def is_abstract_socket_namespace(address: str | bytes | None) -> bool: ...
+
+abstract_sockets_supported: bool = ...
+
+def _remove_temp_dir(rmtree: Callable[[StrOrBytesPath], None], tempdir: StrOrBytesPath) -> None: ...
+def get_temp_dir() -> str: ...
+
+_afterfork_registry: WeakValueDictionary[tuple[int, int, Callable[[Any], Any]], object]
+_afterfork_counter: count[int]
+
+def _run_after_forkers() -> None: ...
+def register_after_fork(obj: object, func: Callable[[Any], Any]) -> None: ...
```

Comment:
I'd prefer to use `Incomplete` for the `obj` parameter and for the parameter of the `func` callable, since it isn't clear to me what those types are meant to be (unless you have a suggestion of how this function is meant to be used? :)

However, I'd prefer to use `object` for the return-type of the `func` callable, since it seems the returned value is ignored at runtime in the one place the functions registered in the `_afterfork_registry` are used: https://github.com/python/cpython/blob/f40bc7fa49f8d137f9b38f7a14569e9a9bdbed07/Lib/multiprocessing/util.py#L167

```suggestion
def register_after_fork(obj: Incomplete, func: Callable[[Incomplete], object]) -> None: ...
```

Reply:
I wasn't sure of the intent, but the reason I chose object was twofold — first, the variable name implied the type of `object` and second, as the value is used with `id` which was the signature `id(__obj: object, /) -> int` my logic was that it made sense to also use `object` for the upstream, as that's the only behavioral requirement.


Happy to change it to incomplete though if you prefer for `obj`

Reply:
`obj` is a fairly common variable name, and doesn't _always_ imply an argument type of `object`; I think I'd prefer to play it safe and use `Incomplete` here :)

Reply:
perfectly fair :) updating and will push that in a moment